pkg/ops/iso.go

```diff
@@ -394,6 +395,13 @@ func (b BuildISOAction) createEFI(rootdir string, isoDir string) error {
 	// Its read from the root of the livecd, so we need to copy it into /EFI/BOOT/grub.cfg
 	// This is due to the hybrid bios/efi boot mode of the livecd
 	// the uefi.img is loaded into memory and run, but grub only sees the livecd root
+	// WARNING: If we load the image into an usb stick, grub will not use the livecd root as the grub root, so it cannot find the grub.cfg
+	// So we copy in 2 places, into the livecd and into the img
+	err = b.cfg.Fs.WriteFile(filepath.Join(temp, constants.EfiBootPath, constants.GrubCfg), []byte(constants.GrubEfiCfg), constants.FilePerm)
+	if err != nil {
+		b.cfg.Logger.Errorf("Failed writing grub.cfg: %v", err)
+		return err
+	}
```

Comment:
could we extract this into a function and just pass it once? I never know what's the best in Go, I find it so verbose, but I know it's common to do this, but it's quite hard for me to see from these lines and the ones below that the only difference is temp and isoDir

Reply:
I think the comment above should be good enough no? even if we set it to a function, the call would be as cryptic as is currently now no?

```
err := WriteGrubCfg(tmpdir)
err = WriteGrubCfg(isodir)
```

Reply:
for me something like

```
err := WriteGrubCfg(temp)
err = WriteGrubCfg(isodir)
```

is always clearer because it really shows only the different arguments passed and not the calculation of paths like here:

```
err = b.cfg.Fs.WriteFile(filepath.Join(temp, constants.EfiBootPath, constants.GrubCfg), []byte(constants.GrubEfiCfg), constants.FilePerm)	
err = b.cfg.Fs.WriteFile(filepath.Join(isoDir, constants.EfiBootPath, constants.GrubCfg), []byte(constants.GrubEfiCfg), constants.FilePerm)	
```

It's simply a lower mental load when reading the code, at least for me coming from the OO world, because you avoid having to also perform these calculations 

```
filepath.Join(temp, constants.EfiBootPath, constants.GrubCfg), []byte(constants.GrubEfiCfg), constants.FilePerm
```

as part of the arguments and if you are interested in them, then you jump into the function making them

Reply:
oki, will do, one sec

Reply:
how about now?

Reply:
oh I think I messed up here, you meant filling the path in the function? But thats just moving the complexity into a different part of the code.

Here in fact, it makes the code more complex because then you have to have 2 different functions for writing the generic config file one for generic, one for ubuntu as their paths are different.
And the functions are just a wrapper for a call to writefile.

I dunno, for me this is worst, having extra functions for a one line makes having to jump around to the function to see what it does. But I dont care either way, I can do it if you want :D